packages/beacon-node/src/metrics/metrics/lodestar.ts

```diff
@@ -718,27 +718,34 @@ export function createLodestarMetrics(
       }),
     },
     recoverDataColumnSidecars: {
-      secFromSlot: register.histogram({
-        name: "lodestar_recover_data_column_sidecar_recover_timer_from_slot",
-        help: "Time elapsed between block slot time and the time data column sidecar recovered",
+      elapsedTimeTillReconstructed: register.histogram({
+        name: "lodestar_data_column_sidecar_elapsed_time_till_reconstructed",
+        help: "Time elapsed between block slot time and the time data column sidecar reconstructed",
         buckets: [2, 4, 6, 8, 10, 12],
       }),
-      recoverTime: register.histogram({
-        name: "lodestar_recover_data_column_sidecar_recover_time_seconds",
-        help: "Time elapsed to recover data column sidecar",
-        // this data comes from 20 blobs in `fusaka-devnet-1`, need to reevaluate in the future
-        buckets: [0.4, 0.6, 0.8, 1.0, 1.2],
+      custodyBeforeReconstruction: register.gauge({
+        name: "lodestar_data_columns_in_custody_before_reconstruction",
+        help: "Number of data columns in custody before reconstruction",
       }),
-      partialColumns: register.gauge({
-        name: "lodestar_recover_data_column_sidecar_partial_columns_total",
-        help: "Total number of partial columns being recovered",
-      }),
-      result: register.gauge<{result: RecoverResult}>({
-        name: "lodestar_recover_data_column_sidecar_recover_result_total",
-        help: "Total count of recover result of data column sidecars",
+      reconstructionResult: register.gauge<{result: RecoverResult}>({
+        name: "lodestar_data_column_sidecars_reconstruction_result",
+        help: "Data column sidecars reconstruction result",
         labelNames: ["result"],
       }),
     },
+    dataColumns: {
+      bySource: register.gauge<{source: DataColumnsSource}>({
+        name: "lodestar_data_columns_by_source",
+        help: "Number of received data columns by source",
+        labelNames: ["source"],
+      }),
+      elapsedTimeTillReceived: register.histogram<{source: DataColumnsSource}>({
+        name: "lodestar_data_column_elapsed_time_till_received",
```

Comment:
same as above, need to add unit

```suggestion
        name: "lodestar_data_column_elapsed_time_till_received_seconds",
```